katsdptelstate/aio/test/test_rdb_writer.py

```diff
@@ -5,49 +5,72 @@
 the async bits.
 """
 
+import logging
 import os
-import shutil
-import tempfile
+from typing import AsyncGenerator
 
-import asynctest
+import pytest
+import pytest_asyncio
 
 import katsdptelstate
 from katsdptelstate import KeyType
 from katsdptelstate.aio import TelescopeState
 from katsdptelstate.aio.rdb_writer import RDBWriter
 
 
-class TestRdbWriter(asynctest.TestCase):
-    async def setUp(self):
-        self.telstate = TelescopeState()
-        await self.telstate.set('immutable', 'abc')
-        await self.telstate.add('mutable', 'def', ts=1234567890.0)
-        await self.telstate.set_indexed('indexed', 'subkey', 'xyz')
-        base_dir = tempfile.mkdtemp()
-        self.filename = os.path.join(base_dir, 'test.rdb')
-        self.addCleanup(shutil.rmtree, base_dir)
-        self.reader = katsdptelstate.TelescopeState()  # Synchronous
-
-    async def test_write_all(self):
-        writer = RDBWriter(self.filename)
-        await writer.save(self.telstate)
+class TestRdbWriter:
+    @pytest_asyncio.fixture
+    async def telstate(self) -> AsyncGenerator[TelescopeState, None]:
+        telstate = TelescopeState()
+        await telstate.set('immutable', 'abc')
+        await telstate.add('mutable', 'def', ts=1234567890.0)
+        await telstate.set_indexed('indexed', 'subkey', 'xyz')
+        yield telstate
+        telstate.backend.close()
+        await telstate.backend.wait_closed()
+
+    @pytest.fixture
+    def reader(self) -> katsdptelstate.TelescopeState:  # Synchronous
+        return katsdptelstate.TelescopeState()
+
+    @pytest.fixture
+    def filename(self, tmpdir) -> str:
```

Comment:
How does `tmpdir` sneak in? Is it a standard fixture?

Reply:
Yes, provided by pytest.